modules/jax/src/main/ts/ephox/jax/core/HttpTypes.ts

```diff
@@ -32,7 +32,7 @@ export interface HttpResponse<T extends ResponseBody> {
 }
 
 export type JwtToken = string;
-export type JwtTokenFactory = (fresh: boolean) => FutureResult<JwtToken, HttpError>;
+export type JwtTokenFactory = <T extends ResponseType>(fresh: boolean) => FutureResult<JwtToken, HttpError<T>>;
```

Comment:
```suggestion
export type JwtTokenFactory<T extends ResponseType> = (fresh: boolean) => FutureResult<JwtToken, HttpError<T>>;
```
I know this means we are going to get more generics propagating everywhere, but I'm pretty sure it has to be this unfortunately. For any given JwtTokenFactory, it's only going to have one type of error it actually returns (based on the response type of the request its based on), so you can't (safely) cast it to a generic function that will return whichever error type the caller wants.

Reply:
Good catch :+1: